.github/workflows/CI.yaml

```diff
@@ -25,12 +27,15 @@ jobs:
       fail-fast: false
       matrix:
         os: [macOS-latest, ubuntu-latest, windows-latest]
-        python-version-min: [3.6, 3.7]
         python-version: [3.7, 3.8, 3.9]
-        
+        env_file: [env, ]
+        include:
```

Comment:
Using `include` we can add extra jobs that aren't part of the main matrix tests. Here we add an extra job that runs python 3.6 on ubuntu and uses the minimum environment yaml file.

Reply:
According to the [Numpy release notes](https://numpy.org/devdocs/release/1.16.0-notes.html), the 1.16.0 version of python is supported in Python 3.5-3.7.
Would you please help me to understand why are we insisting on python=3.6 and Numpy=1.16.0 only? Shouldn't it be tested for 
- Python = 3.6 *And* 3.7 with Numpy=1.16.0?
Similarly, shouldn't we test 
- Python = 3.8  with Numpy=1.18.0,
and
-Python = 3.9 with Numpy=1.19.0?

Reply:
Sorry this is going to be a bit messy to explain :(

So the way I see is that numpy and python are two variables that you can treat separately.

1. It's ok (at least for the sake of sanity) to make the assumption that for any given numpy minor version behaves the exact same way across the multiple python minor versions it supported at the time of release (that's at the very least the assumption that libraries like numpy/scipy and even MDA make). With this in mind, you can just test one version and say it works. 
2. For 1.16.0, here I set it to 3.6 because py3.6 + numpy 1.16.0 will pick up a ton of smaller packages (which the bigger packages we define rely on), that might be lower in version than py3.7 + numpy 1.16.0. In practice this has a near zero impact, but it doesn't hurt.
3. Technically you could test all the numpy versions, but it's very expensive and with libraries like numpy you can kinda make the assumption that as long as the difference between the lowest possible version you can support (1.16.0) and the newest version (1.21.0) isn't too large, then the API has progressively changed between the min & max version. That is to say that anything in 1.21.0 that isn't in 1.16.0 will have been introduced in all the intermediate minor versions. Based on that, you can make the assumption that anything that if things work with the minimum and maximum versions, then everything in between will work fine too.